RubberduckTests/Inspections/ImplicitlyTypedConstInspectionTests.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System.Linq;
+using NUnit.Framework;
+using Rubberduck.CodeAnalysis.Inspections.Concrete;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.VBEditor.SafeComWrappers;
+
+namespace RubberduckTests.Inspections
+{
+    [TestFixture]
+    public class ImplicitlyTypedConstInspectionTests : InspectionTestsBase
+    {
+        [Test]
+        [Category("Inspections")]
+        public void InspectionName()
+        {
+            var inspectionUnderTest = new ImplicitlyTypedConstInspection(null);
+
+            var expected = nameof(ImplicitlyTypedConstInspection);
+            var actual = inspectionUnderTest.Name;
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        [Category("Inspections")]
+        public void ImplicitlyTypedConst_ReturnsResult()
+        {
+            const string inputCode =
+@"Public Sub Foo()
+    Const bar = 0
+End Sub";
+
+            const int expected = 1;
+
+            var results = InspectionResultsForModules(("FooClass", inputCode, ComponentType.ClassModule));
+            var actual = results.Count();
+            Assert.AreEqual(expected , actual);
+        }
+
+        [Test]
+        [Category("Inspections")]
+        public void ImplicitlyTypedConst_DoesNotReturnResult()
+        {
+            const string inputCode =
+@"Public Sub Foo()
+    Const bar As Long = 0
+End Sub";
+
+            const int expected = 0;
+
+            var results = InspectionResultsForModules(("FooClass", inputCode, ComponentType.ClassModule));
+            var actual = results.Count();
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        [Category("Inspections")]
+        [TestCase("!")]
+        [TestCase("@")]
+        [TestCase("#")]
+        [TestCase("$")]
+        [TestCase("%")]
+        [TestCase("^")]
+        [TestCase("&")]
+        public void ImplicitlyTypedConst_HasTypeHint_DoesNotReturnResult(string typeHint)
+        {
+            string inputCode =
+$@"Public Sub Foo()
+    Const bar{typeHint} = 0
+End Sub";
+
+            const int expected = 0;
+
+            var results = InspectionResultsForModules(("FooClass", inputCode, ComponentType.ClassModule));
+            var actual = results.Count();
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        [Category("Inspections")]
+        [TestCase("!")]
+        [TestCase("@")]
+        [TestCase("#")]
+        //[TestCase("$")] Errors out
```

Comment:
What's the error here? If it's breaking the parser, this isn't a problem we can ignore...  `Const foo$ = 0` is a legal declaration AFAICT.

Reply:
The test tests for a type-hint on the RHS, so check for `Const foo = 0$`

Reply:
That test case should be separated. It produces an error because `0` is not a string literal. You can do `"0"$`, but not `0$`.